packages/compose/src/hooks/use-drop-zone/index.js

```diff
@@ -8,39 +8,20 @@ import { useRef } from '@wordpress/element';
  */
 import useRefEffect from '../use-ref-effect';
 
-/* eslint-disable jsdoc/valid-types */
-/**
- * @template T
- * @param {T} value
- * @return {import('react').MutableRefObject<T>} A ref with the value.
- */
-function useFreshRef( value ) {
```

Comment:
@ellatrix is there any difference in behavior between using `useFreshRef` over `useRef( initialValue )`? 

If not I think we can probably remove this helper, especially since it looks like it's tricky to type.

Reply:
We do need useFreshRef. Moving to useRef( initialValue ) instead of ref.current = initialValue breaks related E2E tests.